test_handler.go

```diff
@@ -73,10 +82,32 @@ func testHandler(w http.ResponseWriter, r *http.Request) {
         http.Error(w, err.Error(), http.StatusInternalServerError)
         return
     }
-    testRunsJSON := string(testRunsBytes)
 
-    if err := templates.ExecuteTemplate(w, "index.html", testRunsJSON); err != nil {
+    data := struct {
+        TestRuns string
+        SHA      string
+    }{
+        string(testRunsBytes),
+        runSHA,
+    }
+
+    if err := templates.ExecuteTemplate(w, "index.html", data); err != nil {
         http.Error(w, err.Error(), http.StatusInternalServerError)
         return
     }
 }
+
+func getRunSHA(r *http.Request) (runSHA string, err error) {
+    // Get the SHA for the run being loaded (the first part of the path.)
+    runSHA = "latest"
+    params, err := url.ParseQuery(r.URL.RawQuery)
+    if err != nil {
+        return
```

Comment:
This should be returning `nil, err`

Reply:
Can't have nil for a string, and empty/latest are behaviourally equivalent anyway.

Reply:
Ah got it, but if the function will never return an error then maybe its return value shouldn't be `(runSHA string, err error)`? How about just returning `runSHA string` and adding a docstring describing that `""` or `"latest"` means no SHA was found? A unit test would also be great (doesn't need to be in this PR though).

Reply:
It does return if it errors (more obvious now with explicit return instead of 'naked return')

Reply:
Doc'd and tested.